### DISCUSSION TITLE
Enabling Refactor option in Context menu

### DISCUSSION THREAD

**Discussion by we-wake-act21** [eclipse-theia/theia#12908] [Q&A]
Currently, I am trying to perform refactoring and I'm unable to enable the `Refactor` in context menu.

![image](https://github.com/eclipse-theia/theia/assets/116638890/96841624-e020-47b4-af76-97075e12595d)

**Comment by vince-fugnitto** [comment]
@we-wake-act21 thank you for the discussion, the "refactor" menu item and code-actions are enabled if the [language-server (using LSP)](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/) for your file type supports the request.
